### PR TITLE
plutus-contract: Reduce our use of `row-types`, delete `BlockchainActions`

### PR DESCRIPTION
**Pull request by j-mueller**
* Replace the `BlockchainActions` part of the schema with a pair of types `PABReq` and `PABResp` for the requests/responses that `Contract`s can make
* Apart from reducing our `row-types` usage this also makes it much easier to see what actually gets passed to contract instances
* The schema is now only used for user-defined endpoints
  * Unfortunately the `mkSchema` TH function cannot handle `EmptySchema` so I had to put a dummy endpoint in a couple of places in the playground
* We already had something like this in the PAB, in form of the `ContractPABRequest` and `ContractPABResponse` types. I deleted those.

Pre-submit checklist:
- Branch
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: plutus-contract/src/Plutus/Trace/Emulator/Types.hs [hunk 1]
```diff
@@ -152,17 +155,17 @@ data ContractHandle w s e =
...
 
 instance Pretty EmulatorRuntimeError where
     pretty = \case
-        ThreadIdNotFound i   -> "Thread ID not found:" <+> pretty i
-        InstanceIdNotFound w -> "Instance ID not found:" <+> pretty w
-        JSONDecodingError e  -> "JSON decoding error:" <+> pretty e
-        GenericError e       -> pretty e
+        ThreadIdNotFound i            -> "Thread ID not found:" <+> pretty i
+        InstanceIdNotFound w          -> "Instance ID not found:" <+> pretty w
+        EmulatorJSONDecodingError e v -> "emulator JSON decoding error:" <+> pretty e <+> parens (viaShow v)
```
Comment (koslambrou) [inline plutus-contract/src/Plutus/Trace/Emulator/Types.hs:167]: Capitalise?

File: plutus-contract/src/Plutus/Contract.hs [hunk 2]
```diff
@@ -90,50 +72,23 @@ module Plutus.Contract(
...
-  , HasWriteTx s
-  , HasUtxoAt s
-  , HasOwnPubKey s
-  , HasTxConfirmation s
-  , HasOwnId s
-  )
+import qualified Control.Monad.Freer.Extras.Log as L
+import qualified Control.Monad.Freer.Writer     as W
+import           Ledger.AddressMap              (UtxoMap)
+import           Prelude                        hiding (until)
```
Comment (koslambrou) [inline plutus-contract/src/Plutus/Contract.hs:89]: `hiding (until)` not needed anymore

**Review by koslambrou — APPROVED** [review]

**Comment by koslambrou** [comment]
LGTM. Curiously, modules such as `Plutus.Contract.Effects.AwaitSlot` and `Plutus.Contract.Effects.AwaitTxConfirmed` were almost completely stripped of it's content, but they still exist. Why's that?

**Comment by j-mueller** [comment]
> but they still exist. Why's that?

I meant to delete them when I was done but I forgot. Thanks for pointing it out!

File: plutus-contract/test/Spec/Contract.hs [hunk 3]
```diff
@@ -199,7 +189,7 @@ tests =
               matchLogs :: [EM.EmulatorTimeEvent ContractInstanceLog] -> Bool
               matchLogs lgs =
                   case (_cilMessage . EM._eteEvent <$> lgs) of
-                            [ Started, ContractLog "waiting for endpoint 1", CurrentRequests [_], ReceiveEndpointCall _, ContractLog "Received value: 27", HandledRequest _, CurrentRequests [], StoppedNoError] -> True
+                            [ Started, ContractLog "waiting for endpoint 1", CurrentRequests [_], ReceiveEndpointCall{}, ContractLog "Received value: 27", HandledRequest _, CurrentRequests [], StoppedNoError] -> True
```
Comment (koslambrou) [inline plutus-contract/test/Spec/Contract.hs:192]: waiting => Waiting

**Review by koslambrou — APPROVED** [review]

File: plutus-contract/test/Spec/Contract.hs (hunk 3 above)
Comment (j-mueller) [reply plutus-contract/test/Spec/Contract.hs:192]: oh yes of course...

Comment (j-mueller) [reply plutus-contract/test/Spec/Contract.hs:192]: hm no it passes locally?

**Pull request merged**